### ISSUE TITLE
--clientname option seems broken in v1.2

### ISSUE THREAD

**Issue by dyfer** [jacktrip/jacktrip#73]
I'm using a binary build from https://ccrma.stanford.edu/software/jacktrip/osx/index.html

I'm having a [problem with long hostnames](https://groups.google.com/forum/#!topic/jacktrip-users/aJvGS-ykpBQ) when using JackTrip v1.2. I _think_ the problem might be traced to using the (long) hostname as JACK client name. To circumvent that I tried to set custom client name and this seems to not work in v 1.2.

It seems that provided `--clientname` is overwritten in https://github.com/jacktrip/jacktrip/blob/master/src/JackTrip.cpp#L164 

It is unclear to me what WAIR (Wide Area Internet Reverberation?) is exactly in terms of JackTrip functionality so I'm not sure what the proper solution is. I'd guess that if custom client name is provided, it shouldn't be overwritten in the code linked above, however I'm not sure how to best achieve that.

**Comment by psiborg112** [comment]
Hi Marcin,
Just wanted to check, how many characters long is your server's name? 
And I'm assuming you're running Jack 2 because of the c++ style error 
codes you're getting?

There are two issues here - custom client names don't seem to be 
honoured at all at the moment and we should either make sure they are or 
remove the option altogether. (The only time they should be overwritten 
is in hub server mode.) The other is that we should check the hostname 
length and limit the derived client name length as appropriate.

Should be an easy fix, and I'll take a look at it today or tomorrow.
Cheers,
Aaron

**Comment by dyfer** [comment]
Hi Aaron!
Yes, I'm using JACK 2 from the JackOSX package.
My server's name is 32 characters. However, I was trying to limit the size in `virtual void setClientName(const char* ClientName)` inside `JackAudioInterface.h` and it seems that `jack_client_name_size();` returns 65... However this **worked** (not on every run though...) and I was able to connect to the server.
Here's what I used (`JackAudioInterface.h`):
```cpp
    virtual void setClientName(const char* ClientName)
    { 
      std::string newName(ClientName);
      size_t maxSize = jack_client_name_size();
      // if(newName.length() > maxSize) //commented out since the size is actually not larger than the limit...
        newName.resize(maxSize);

      mClientName = newName.c_str(); 
    }
```
IMO the option for a custom client name is quite important to keep. In the current setup I'm splitting multichannel stream into multiple instances of JackTrip to get around packet size limitation and it would be problematic to not have custom names for them.

**Comment by psiborg112** [comment]
Thanks for that. It's weird that it's any sort of issue when your name 
is well below the limit. I'll try to look into it further. In the 
meantime, if you replace lines 162-164 in JackTrip.cpp with the 
following, it should honour custom client names. (And I'll submit a pull 
request with these changes.)

if(mPeerAddress.toStdString()!="" && (mJackClientName == "JackTrip" || 
mJackTripMode == SERVERPINGSERVER)) {
     mJackClientName = QString(mPeerAddress).replace(":", 
".").toLatin1().constData();
}

Cheers,
Aaron

**Comment by dyfer** [comment]
Thanks Aaron!

It seems that the limit on my system is 27 (?!) characters for the hostname:
- fail
```sh
-> jacktrip -c 123456789.123456789.12345678
mPeerAddress "123456789.123456789.12345678" false
WAIR ID 0 jacktrip client name set to=123456789.123456789.12345678
Cannot open 123456789.123456789.12345678 client
JackShmReadWritePtr1::~JackShmReadWritePtr1 - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ERROR:
Maybe the JACK server is not running?
Exiting JackTrip...
---------------------------------------------------------
```
- pass:
```sh
-> jacktrip -c 123456789.123456789.1234567
mPeerAddress "123456789.123456789.1234567" false
WAIR ID 0 jacktrip client name set to=123456789.123456789.1234567
Jack: JackClient::SetupDriverSync driver sem in flush mode
Jack: JackPosixSemaphore::Connect name = js_123456789.123456789.1234567
Jack: JackPosixSemaphore::Connect sem_getvalue 0
Jack: JackLibClient::Open name = 123456789.123456789.1234567 refnum = 3
Jack: JackClient::PortRegister ref = 3 name = 123456789.123456789.1234567:send_1 type = 32 bit float mono audio port_index = 13
Jack: JackClient::PortRegister ref = 3 name = 123456789.123456789.1234567:send_2 type = 32 bit float mono audio port_index = 14
Jack: JackClient::PortRegister ref = 3 name = 123456789.123456789.1234567:receive_1 type = 32 bit float mono audio port_index = 15
Jack: JackClient::PortRegister ref = 3 name = 123456789.123456789.1234567:receive_2 type = 32 bit float mono audio port_index = 16
Setting JACK Process Callback...
SUCCESS
---------------------------------------------------------
The Sampling Rate is: 48000
---------------------------------------------------------
The Audio Buffer Size is: 128 samples
                      or: 512 bytes
---------------------------------------------------------
The Number of Channels is: 2
---------------------------------------------------------
The RTAudio device ID is: 0
---------------------------------------------------------
Using UDP Protocol
---------------------------------------------------------
ERROR:
Incorrect presentation format address
 '123456789.123456789.1234567' is not a valid IP address or Host Name
Exiting JackTrip...
---------------------------------------------------------
```

**Comment by psiborg112** [comment]
Thanks for testing that Marcin. Good to know where the limit is even if 
I have no idea why that would be the case.
Cheers,
Aaron

**Comment by dyfer** [comment]
Hi @psiborg112 
I noticed you closed the PR that would solve this. Are you planning to open another PR for this? Otherwise I'd be happy to submit a patch.

**Comment by psiborg112** [comment]
Hi Marcin,
I closed it because I stupidly started adding some much bigger changes 
to the branch that I'd done this in. (Should have kept it in a separate 
branch.) I'll recreate the fix in a new branch now.
Cheers,
Aaron

**Comment by dyfer** [comment]
Thanks, Aaron!

**Comment by ntonnaett** [comment]
What is missing that we can see this as completely fixed?

**Comment by dyfer** [comment]
This has been fixed in #93. Thank you!

**Comment by psiborg112** [comment]
The only thing that was missing was the long Jack client name issue (which is probably a Jack issue, but something we might consider working around or at least giving some sort of warning or user feedback about), and that's been moved to #102
